### DISCUSSION TITLE
Not sure I understand replication correctly

### DISCUSSION THREAD

**Discussion by Stitch10925** [seaweedfs/seaweedfs#2899] [Q&A] [answered]
I'm my test setup I have 1 master and 2 machines each holding 8 volumes, both in the same rack. Default replicating was set to 1 copy in the same rack.

From the documentation I gathered that an upload will only complete if all replication is finished. So I wondered what would happen if I kill one of the volumes while uploading a large file.

It seems that the upload stops, but then seems to continue at a very low speed and in waves (slow upload, pause, slow upload, pause, ...). Eventually the upload would finish, which seems to go against the earlier statement that all replications need to be successful.

When I then enabled the 2nd volume again it would start replicating the file and growing the volumes. However, the file was never accessible or visible.

I'm not quite sure I understand the replication and how resilient it is. Can someone explain what was going on during the "wave" period and why the file never became available?

**Comment by chrislusf** [comment, marked as answer]
You would need to have more than 2 machines with 001.

**Reply by Stitch10925** [reply]
How about with EC? Is it also a minimum of 3 machines?

**Reply by chrislusf** [reply]
EC still can encode and decode. But with 14 shards, losing one server out of 3 will lose the data. Minimum is 4 servers.

**Reply by Stitch10925** [reply]
Thank you for the fast reply and information
